src/app/locator/qgsinbuiltlocatorfilters.cpp

```diff
@@ -702,3 +708,205 @@ void QgsBookmarkLocatorFilter::triggerResult( const QgsLocatorResult &result )
+
+      bool withinWgs84 = wgs84Crs.bounds().contains( point );
+      if ( currentCrs != wgs84Crs )
+      {
+        QgsLocatorResult result;
+        result.filter = this;
+        result.displayString = tr( "Go to %1 %2 (Map CRS, %3)" ).arg( QString::number( point.x(), 'g', 10 ), QString::number( point.y(), 'g', 10 ), currentCrs.userFriendlyIdentifier() );
+        result.userData = data;
+        result.score = 0.9;
+        emit resultFetched( result );
+      }
+
+      if ( withinWgs84 )
+      {
+        if ( currentCrs != wgs84Crs )
+        {
+          QgsCoordinateTransform transform( wgs84Crs, currentCrs, QgsProject::instance()->transformContext() );
+          QgsPointXY transformedPoint;
+          try
+          {
+            transformedPoint = transform.transform( point );
+          }
+          catch ( const QgsException &e )
+          {
+            Q_UNUSED( e )
+            return;
+          }
+          data[QStringLiteral( "point" )] = transformedPoint;
+        }
+
+        QgsLocatorResult result;
+        result.filter = this;
+        result.displayString = tr( "Go to %1° %2° (%3)" ).arg( QString::number( point.x(), 'g', 10 ), QString::number( point.y(), 'g', 10 ), wgs84Crs.userFriendlyIdentifier() );
+        result.userData = data;
+        result.score = 1.0;
+        emit resultFetched( result );
+      }
+    }
+    return;
+  }
+
+  QMap<int, double> scales;
+  scales[0] = 739571909;
+  scales[1] = 369785954;
+  scales[2] = 184892977;
+  scales[3] = 92446488;
+  scales[4] = 46223244;
+  scales[5] = 23111622;
+  scales[6] = 11555811;
+  scales[7] = 5777905;
+  scales[8] = 2888952;
+  scales[9] = 1444476;
+  scales[10] = 722238;
+  scales[11] = 361119;
+  scales[12] = 180559;
+  scales[13] = 90279;
+  scales[14] = 45139;
+  scales[15] = 22569;
+  scales[16] = 11284;
+  scales[17] = 5642;
+  scales[18] = 2821;
+  scales[19] = 1500;
+  scales[20] = 1000;
+
+  QUrl url( string );
+  if ( url.isValid() )
+  {
+    double scale = 0.0;
+    bool okX = false;
+    bool okY = false;
+    double posX = 0.0;
+    double posY = 0.0;
+    if ( url.hasFragment() )
+    {
+      // Check for OSM/Leaflet/OpenLayers pattern (e.g. http://www.openstreetmap.org/#map=6/46.423/4.746)
+      QStringList fragments = url.fragment().split( '&' );
+      for ( const QString &fragment : fragments )
+      {
+        if ( fragment.startsWith( QStringLiteral( "map=" ) ) )
+        {
+          QStringList params = fragment.mid( 4 ).split( '/' );
+          if ( params.size() >= 3 )
+          {
+            if ( scales.contains( params.at( 0 ).toInt() ) )
+            {
+              scale = scales.value( params.at( 0 ).toInt() );
+            }
+            posX = params.at( 2 ).toDouble( &okX );
+            posY = params.at( 1 ).toDouble( &okY );
+          }
+          break;
+        }
+      }
+    }
+
+    if ( !okX && !okY && string.indexOf( QStringLiteral( "google.com/maps/" ) ) != -1 )
```

Comment:
It would be nice to not limit this to `.com` but also handle international TLDs

Reply:
@uclaros , URL example?

Reply:
No worries, got i, will adjust accordingly. Good catch.